### PR TITLE
[fix](date)difference between BE and FE in determining a leap year.

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

fe code

```java
public static boolean isLeap(long year) {
    return ((year & 3) == 0) && ((year % 100) != 0 || (year % 400) == 0);
}
```

be code

```C++
bool is_leap(uint32_t year) {
    return ((year % 4) == 0) && ((year % 100 != 0) || ((year % 400) == 0 && year));
}
```

where the year is not 0.

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

**Comment by Mryange** [comment]
run buildall

**Review by zclllyybb — COMMENTED** [review]
add ut please

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

File: fe/fe-core/src/test/java/org/apache/doris/nereids/rules/expression/FoldConstantTest.java [hunk 1]
```diff
@@ -524,6 +525,19 @@ void testDateV2TypeDateTimeArithmeticFunctions() {
         Assertions.assertEquals(DateTimeExtractAndTransform.lastDay(dateLiteral).toSql(), answer[answerIdx]);
     }
 
+    @Test
+    void testDateError() {
+        System.out.println("error data test 0000-02-29");
```
Comment (morrySnow) [inline fe/fe-core/src/test/java/org/apache/doris/nereids/rules/expression/FoldConstantTest.java:530]: remove this print

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run p0

**Review by zclllyybb — APPROVED** [review]

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run feut

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**